jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -629,21 +626,13 @@ public String toConnectionString()
         public Content(ByteBuffer content)
         {
             super(content);
-            _contentBufferReferences.incrementAndGet();
+            _retainableByteBuffer.retain();
         }
 
         @Override
         public void succeeded()
         {
-            int counter = _contentBufferReferences.decrementAndGet();
-            if (counter == 0)
-                releaseRequestBuffer();
-            // TODO: this should do something (warn? fail?) if _contentBufferReferences goes below 0
-            if (counter < 0)
-            {
-                LOG.warn("Content reference counting went below zero: {}", counter);
-                _contentBufferReferences.incrementAndGet();
-            }
+            _retainableByteBuffer.release();
```

Comment:
The safety code was removed, but I would keep it.

Reply:
This safety code is now in `RetainableByteBuffer.release()`